src/core/ai/llm/validation.test.ts

```diff
@@ -0,0 +1,417 @@
+import { describe, test, expect, vi, beforeEach } from 'vitest';
+import { validateInputForLLM, createInputValidationError } from './validation.js';
+import * as registry from './registry.js';
+
+// Mock the registry module
+vi.mock('./registry.js', () => ({
+    validateModelFileSupport: vi.fn(),
+    getAllowedMimeTypes: vi.fn(),
+}));
+
+describe('validateInputForLLM', () => {
+    const mockValidateModelFileSupport = vi.mocked(registry.validateModelFileSupport);
+    const mockGetAllowedMimeTypes = vi.mocked(registry.getAllowedMimeTypes);
+
+    beforeEach(() => {
+        vi.clearAllMocks();
+        // Default mock implementations
+        mockGetAllowedMimeTypes.mockReturnValue(['application/pdf', 'audio/mp3', 'audio/wav']);
+        mockValidateModelFileSupport.mockReturnValue({
+            isSupported: true,
+        });
```

Comment:
@coderabbitai does it make sense to mock these functions - my thought was that we shoudl just use the direct registry functions and only mock for edge-cases where we don't have valid examples, wdyt? mocking these doesn't seem helpful to me (we test mock behaviour rather than real behaviour)

Reply:
ok i will handle this in https://github.com/truffle-ai/saiki/pull/249/files, can you suggest the tests